### PR TITLE
Added timezone module

### PR DESCRIPTION
**Pull request by tmshn**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME
- timezone
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel 1fc44e4103) last updated 2016/05/02 10:56:19 (GMT +900)
  lib/ansible/modules/core: (detached HEAD b6ad3b6773) last updated 2016/05/02 10:56:20 (GMT +900)
  lib/ansible/modules/extras: (detached HEAD 1846de2809) last updated 2016/05/02 10:56:20 (GMT +900)
  config file =
  configured module search path = Default w/o overrides
```
##### SUMMARY

This module configures the timezone of the system (for only Linux systems).
It can modify both system clock and hardware clock (although the latter is not recommended).

With the below task,

``` yaml
tasks:
  - name: set timezone to JST
    timezone: name=Asia/Tokyo
```

The verbose output with the `--diff` option would be:

``` diff
changed: [default] => {"changed": true, "msg": "executed `/usr/bin/timedatectl set-timezone Asia/Tokyo`"}
--- before
+++ after
@@ -1,3 +1,3 @@
 {
-    "name": "UTC"
+    "name": "Asia/Tokyo"
 }
```
##### NOTE

This is the new, improved version of my old PR #2051

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @tmshn for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: system/timezone.py [hunk 1]
```diff
@@ -0,0 +1,458 @@
...
+      description: The values before change
+      type: dict
+    after:
+      description: The values after change
+      type: dict
+'''
+
+EXAMPLES = '''
+- name: set timezone to Asia/Tokyo
+  timedatectl: name=Asia/Tokyo
```
Comment (alxgu) [inline system/timezone.py]: Is timedatectl correct? The module is called timezone, isn't it?

**Comment by alxgu** [comment]
A hint in the documentation that you need to restart `crond` by timezone changes, would be nice.

File: system/timezone.py (hunk 1 above)
Comment (tmshn) [inline system/timezone.py]: Ah sorry, that's my mistake. I will modify.

**Comment by tmshn** [comment]
@alxgu OK, will add that point to the docs.

Can I do `git rebase -i` for my branch?

**Comment by resmo** [comment]
this PR seems to need revision:
needs_revision

**Comment by gregdek** [comment]
Thanks @tmshn for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by tmshn** [comment]
Revised and now ready_for_review 👍

**Comment by gregdek** [comment]
Thanks @tmshn for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by raben2** [comment]
shipit

**Comment by skornehl** [comment]
shipit

**Comment by alxgu** [comment]
@tmshn How would you extend the module for further distrubtions. Fore example SLES.
The config file is also `/etc/sysconfig/clock` but the keyword is `TIMEZONE=Europe/Berlin`

**Comment by alikins** [comment]
Looks reasonably extensible to me. Got two shipits. Shipping it.

**Comment by tmshn** [comment]
@alxgu The pr been merged, I leave it to the community to extend this module to other platforms (of course I'll do if I can afford).

**Pull request merged**